src/components/plot/bookmarkbutton.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
+import {FacetedCompositeUnitSpec} from 'vega-lite/build/src/spec';
+import {BOOKMARK_ADD_PLOT, BOOKMARK_REMOVE_PLOT, BookmarkAction} from '../../actions/bookmark';
+import {ActionHandler} from '../../actions/redux-action';
+import {Bookmark} from '../../models/bookmark';
+import {PlotFieldInfo} from '../../models/plot';
+import * as styles from './bookmarkbutton.scss';
+
+
+
+export interface BookmarkProps extends ActionHandler<BookmarkAction> {
+  bookmark: Bookmark;
+  plotObjectFieldInfos: PlotFieldInfo[];
+  plotObjectSpec: FacetedCompositeUnitSpec;
+}
+
+export class BookmarkButtonBase extends React.PureComponent<BookmarkProps, any> {
+  constructor(props: BookmarkProps) {
+    super(props);
+
+    this.state = {
+      isBookmarked: !!this.props.bookmark.dict[JSON.stringify(this.props.plotObjectSpec)], openDialog: false
+    };
+
+    this.onBookmarkRemove = this.onBookmarkRemove.bind(this);
+    this.onBookmarkAdd = this.onBookmarkAdd.bind(this);
+    this.onBookmarkClick = this.onBookmarkClick.bind(this);
+    this.onKeepBookmark = this.onKeepBookmark.bind(this);
+  }
+
+  public render() {
+    const bookmarkColor = (this.state.isBookmarked) ? '#0c96d0' : '#aaa';
+    const openDialog = (this.state.openDialog) ? 'block' : 'none';
+
+    let bookmarkAlert;
+    if (this.state.openDialog) {
+      bookmarkAlert = (
+        <div styleName="bookmarkAlert" style={{display: openDialog}}>
```

Comment:
The `bookmarkAlert` variable is unused?   (Please review your own PR before asking me to review.)